crypto/fipsmodule/rand/new_rand_prefix.h

```diff
@@ -0,0 +1,21 @@
+// Prefix some symbols.
+// Delete when promoting "new rand" to default choice.
+
+#ifndef OPENSSL_HEADER_CRYPTO_RAND_LOCAL_PREFIX_INTERNAL_H
+#define OPENSSL_HEADER_CRYPTO_RAND_LOCAL_PREFIX_INTERNAL_H
+
+#include <openssl/base.h>
+
+#define NR_CONCAT(x,y) x##_##y
+#define NR_PREFIX(x) NR_CONCAT(new_rand,x)
+
+#endif // OPENSSL_HEADER_CRYPTO_RAND_LOCAL_PREFIX_INTERNAL_H
+
+#if defined(COMPILATION_UNIT_NR_PREFIX)
+
+#define RAND_bytes_with_additional_data NR_PREFIX(RAND_bytes_with_additional_data)
```

Comment:
Does this actually cause everything in aws-lc to flip over to the new random code? This could be potentially confusing to debug what version is being used based on when this header file gets included some code could use the real/old version, and some code could use the new version.

Reply:
I can see why this file is a bit confusing. It's needed such that I can use the existing function names in e.g. testing code.

The `OPENSSL_HEADER_CRYPTO_RAND_LOCAL_PREFIX_INTERNAL_H` part ensures I can use the prefix macros where I need themm.

Now, note that you have to actively define `COMPILATION_UNIT_NR_PREFIX` for something to happen. I will use this part in compilation units where I need to write e.g. `RAND_bytes` many times. For example in tests. When we flip over the default to the new randomness generation I won't have to go around and change names at 1000 places. I just delete this prefix-file from the compilation unit. It is also a way where I can flip over existing tests to use my new implementation during development.

I don't use the part `COMPILATION_UNIT_NR_PREFIX` everywhere because of how our FIPS module build works. To ensure that my stuff works for our fips build I need to add e.g. `new_rand.c` to `bcm.c`. But after pre-processing that is just one giant file. Therefore, I can't even use the same struct type names. Even static functions with the same name would result in redefinition errors.

The prefix build also make things complicated. Because the prefix macros are included everywhere.